src/game/objects/general/door.c

```diff
@@ -51,12 +51,12 @@ static void Door_Shut(DOORPOS_DATA *const d)
         return;
     }
 
-    sector->index = 0;
     sector->box = NO_BOX;
     sector->floor.height = NO_HEIGHT;
     sector->ceiling.height = NO_HEIGHT;
     sector->portal_room.sky = NO_ROOM;
     sector->portal_room.pit = NO_ROOM;
+    sector->portal_room.wall = NO_ROOM;
```

Comment:
Is it safe to introduce this assignment?

Reply:
Ah, apologies, I forgot to discuss this above. This was the last reference to the sector's index, along with in `Room_GetSector`, and I overlooked it in the last PR. When we had `Room_GetDoor`, setting the sector's index to 0 meant that that returned `NO_ROOM` when the door is shut. So this is just doing the same but more explicitly. When the door is opened, the original sector gets restored.
I think this "FD nulling" is in place otherwise Lara could glitch through closed doors and then `Room_GetSector` would push her through the portal.